src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java

```diff
@@ -62,6 +62,8 @@ private AutosleepServiceInstance() {
 
     public AutosleepServiceInstance(CreateServiceInstanceRequest request) throws HttpMessageNotReadableException {
         super(request);
+        //only set secret on creation
+        setSecretFromParams(request.getParameters());
```

Comment:
Yeah I know: that's why it is not mandatory in the setter. I only set it in a create context (with Create Request). The authorization only compare the one in the request with the one in the object